include/pulsar/Producer.h

```diff
@@ -23,13 +23,16 @@
 #include <pulsar/defines.h>
 #include <stdint.h>
 
+#include <functional>
 #include <memory>
 
 namespace pulsar {
 class ProducerImplBase;
 class PulsarWrapper;
 class PulsarFriend;
 
+typedef std::function<void(Result, const MessageId& messageId)> SendCallback;
+typedef std::function<void(Result)> CloseCallback;
```

Comment:
Why did you add these extra definitions?

Reply:
Fixed by https://github.com/apache/pulsar-client-cpp/pull/169#discussion_r1066689463